assets/app/views/editor/edit-file.js

```diff
@@ -0,0 +1,159 @@
+var fs = require('fs');
+
+var Backbone = require('backbone');
+var _ = require('underscore');
+
+var toMarkdown = require('to-markdown');
+var markdown = require('markdown').markdown;
+
+var SirTrevor = require('sir-trevor');
+
+// var Block = require('../models/Block').model;
+// var Blocks = require('../models/Block').collection;
+
+var headings = require('./blocks/heading');
+SirTrevor.Blocks.H1 = headings.h1;
+SirTrevor.Blocks.H2 = headings.h2;
+SirTrevor.Blocks.H3 = headings.h3;
+SirTrevor.Blocks.Ordered = require('./blocks/ol');
+SirTrevor.Blocks.Unordered = require('./blocks/ul');
+SirTrevor.Blocks.Code = require('./blocks/code');
+
+var templateHtml = fs.readFileSync(__dirname + '../../templates/editor/edit-file.html').toString();
+
+var EditorView = Backbone.View.extend({
+  tagName: 'div',
+  events: {
+    'click #save-content-action': 'saveDocument'
+  },
+  initialize: function (opts) {
+    this.doc = extractFrontMatter(opts.content);
+    this.file = opts.file;
+    return this;
+  },
+  render: function () {
+    var doc     = this.doc,
+        blocks  = [],
+        editor, mdTree;
+
+    this.$el.html(_.template(templateHtml)({ file: this.file }));
+    this.editor = new SirTrevor.Editor({
+      el: this.$('.js-st-instance'),
+      blockTypes: ["H1", "H2", "H3", "Text", "Unordered", "Ordered"]
+    });
+
+    mdTree = markdown.parse(doc.content).slice(1);
+    for (var i = 0; i < mdTree.length; i++) {
+      blocks.push(jsonMLToBlock(mdTree[i]));
+    }
+
+    this.$('.js-st-instance').text(JSON.stringify({ data: blocks }));
+    this.editor.reinitialize();
+
+    return this;
+  },
+  saveDocument: function (e) {
+    var msg = this.$('#save-content-message').val(),
```

Comment:
I'd probably be consistent and use one way to select elements form JS. I'd say prefixed classes ("js-") or data attributes.

This is also because the way I build components, I rarely envision them being used just once on a page, I don't like that constraint.